proto/storage/config.proto

```diff
@@ -79,9 +79,22 @@ message PrivateConfig {
   AdministrationEventsConfig administration_events_config = 8;
 }
 
+message PlatformComponentConfig {
+  message Rule {
+    message NamespaceRule {
+      string regex = 1;
```

Comment:
Do we need regex here? Why not globs or list of string for exact match?
In my experience giving user ability to write regex ends in weird problems starting from usability (what regex dialect do we support) and ending on performance. Feel free to ignore if that's not the case here

Reply:
This does bring up one key thought.  When we document this we should probably call out things we do not support that is limited by the `go` library we use.

Reply:
The list could get pretty long even if just openshift namespaces are considered, which is why the choice of using regex here. But I do agree that we need to document that we only support go regex.

Reply:
Regardless of whether we decide to keep the regex, this proto seems over engineered. For consideration:

message PlatformComponentConfig {
message Rule {
string name = 1;
string regex = 2;
}
repeated Rule namespace_rules = 1;
bool needs_reevaluation = 2;
}

Reply:
It was my suggestion to add some extra structure to the API to make it more expandable.  More context here: https://redhat-internal.slack.com/archives/C07EJJ8K8R5/p1742871728318989?thread_ts=1742833731.792429&cid=C07EJJ8K8R5.